orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/tasks/RegisterCanaryTask.groovy

```diff
@@ -59,21 +61,30 @@ class RegisterCanaryTask implements Task {
       if (response.status == 200 && response.body.mimeType().startsWith('text/plain')) {
         canaryId = response.body.in().text
       }
-    } catch (RetrofitError re) {
+    } catch (SpinnakerHttpException re) {
       def response = [:]
       try {
-        def responseBody = re.getBodyAs(Map) as Map
+        def responseBody = re.responseBody
         response = responseBody!=null ? responseBody : response
       } catch (Exception e) {
         response.error = e.message
       }
 
-      response.status = re.response?.status
-      response.errorKind = re.kind
+      response.status = re.responseCode
+      response.errorKind = "HTTP"
 
       throw new IllegalStateException(
         "Unable to register canary (executionId: ${stage.execution.id}, stageId: ${stage.id} canary: ${c}), response: ${response}"
       )
+    } catch(SpinnakerServerException e){
+      if(e instanceOf SpinnakerNetworkException) {
+        def response = [:]
+        response.errorKind = "NETWORK"
+        response.status = null
+        throw new IllegalStateException(
```

Comment:
Doesn't the old code do this for all RetrofitErrors, not only network errors?

Reply:
The wiremock tests could only replicate the network and http errors, So i implemented in this way.

However, this part of the code is modified in the recent commits. Default case is also now added.